com.hazelcast.jet.pipeline.test.TestSourcesTest.test_itemStream_withWindowing[0: mode=member]

_master_ + PR https://github.com/hazelcast/hazelcast-jet/pull/1459 (commit 8e11f69da2c8f84137601c68b1009883416b44db)

Failed on PR builder (but very probably not related to changes): http://jenkins.hazelcast.com/job/jet-oss-pr-builder/669/testReport/junit/com.hazelcast.jet.pipeline.test/TestSourcesTest/test_itemStream_withWindowing_0__mode_member_/

Stacktrace:
```
Expected: an instance of com.hazelcast.jet.pipeline.test.AssertionCompletedException
     but: <java.lang.AssertionError: java.lang.AssertionError: expected:<10> but was:<9>> is a java.lang.AssertionError
Stacktrace was: java.lang.AssertionError: java.lang.AssertionError: expected:<10> but was:<9>
 at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:178)
 at com.hazelcast.jet.function.ConsumerEx.accept(ConsumerEx.java:43)
 at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67)
 at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60)
 at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:64)
 at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:287)
 at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:270)
 at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:242)
 at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:327)
 at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:891)
 at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:308)
 at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.AssertionError: expected:<10> but was:<9>
 at org.junit.Assert.fail(Assert.java:88)
 at org.junit.Assert.failNotEquals(Assert.java:834)
 at org.junit.Assert.assertEquals(Assert.java:645)
 at org.junit.Assert.assertEquals(Assert.java:631)
 at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_itemStream_withWindowing$f81069ef$1(TestSourcesTest.java:80)
 at com.hazelcast.jet.function.ConsumerEx.accept(ConsumerEx.java:43)
 at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:167)
 at com.hazelcast.jet.function.BiConsumerEx.accept(BiConsumerEx.java:43)
 at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75)
 at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:188)
 at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:597)
 at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109)
 at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:54)
 at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:165)
 at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:300)
 ... 6 more
```

Standard output:
```
Hiccups measured while running test 'test_itemStream_withWindowing[0: mode=member](com.hazelcast.jet.pipeline.test.TestSourcesTest):'
07:04:55, accumulated pauses: 171 ms, max pause: 111 ms, pauses over 1000 ms: 0
07:05:00, accumulated pauses: 216 ms, max pause: 177 ms, pauses over 1000 ms: 0
07:05:05, accumulated pauses: 43 ms, max pause: 5 ms, pauses over 1000 ms: 0
07:05:10, accumulated pauses: 1 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java
@@ -75,9 +75,9 @@ public class TestSourcesTest extends PipelineTestSupport {
          .window(WindowDefinition.tumbling(1000))
          .aggregate(AggregateOperations.counting())
          .apply(assertCollectedEventually(10, windowResults -> {
-             assertTrue("sink list should contain some items", windowResults.size() > 1);
-             // first window may be incomplete, subsequent windows should have 10 items
-             assertEquals(10L, (long) windowResults.get(1).result());
+             // find any window that has 10 items, some may be incomplete due to hiccups
+             boolean matched = windowResults.stream().anyMatch(r -> r.result() == itemsPerSecond);
+             assertTrue("Did not find any window with 10 items: " + windowResults, matched);
          }));
 
         expectedException.expect(AssertionCompletedException.class);

